src/stacktrace.rs

```diff
@@ -0,0 +1,247 @@
+extern crate lazy_static;
+
+use crate::error::*;
+use gdb_command::stacktrace::*;
+use regex::Regex;
+use std::collections::HashSet;
+use std::fmt;
+use std::io::Write;
+use std::process::{Command, Stdio};
+use std::sync::RwLock;
+
+lazy_static::lazy_static! {
+    // Regular expressions for functions to be ignored.
+    pub static ref STACK_FRAME_FUNCTION_IGNORE_REGEXES: RwLock<Vec<String>> = RwLock::new(
+        Vec::new());
+    // Regular expressions for file paths to be ignored.
+    pub static ref STACK_FRAME_FILEPATH_IGNORE_REGEXES: RwLock<Vec<String>> = RwLock::new(
+        Vec::new());
+}
+
+/// Information about line in sources which caused a crash.
+pub enum CrashLine {
+    // source:line:column.
+    Source(DebugInfo),
+    // Binary module and offset.
+    Module { file: String, offset: u64 },
+}
+
+impl fmt::Display for CrashLine {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match &self {
+            CrashLine::Source(debug) => {
+                if debug.line != 0 && debug.column != 0 {
+                    write!(f, "{}:{}:{}", debug.file, debug.line, debug.column)
+                } else if debug.line != 0 {
+                    write!(f, "{}:{}", debug.file, debug.line)
+                } else {
+                    write!(f, "{}", debug.file)
+                }
+            }
+            CrashLine::Module { file, offset } => {
+                write!(f, "{file}+{offset:#x}")
+            }
+        }
+    }
+}
+
+pub trait ParseStacktrace {
+    /// Extract stack trace from stream
+    fn extract_stacktrace(stream: &str) -> Result<Vec<String>>;
+
+    /// Transform stack trace strings into Stacktrace type
+    fn parse_stacktrace(entries: &[String]) -> Result<Stacktrace>;
+
+    /// Get crash line from stack trace: source:line or binary+offset.               
+    fn crash_line(stacktrace: &Stacktrace) -> Result<CrashLine> {
```

Comment:
Create CrashLineExt trait for casr::Stacktrace and implement it.

Reply:
done